dartagnan/src/main/java/com/dat3m/dartagnan/encoding/WmmEncoder.java

```diff
@@ -636,11 +635,10 @@ public Void visitSyncBarrier(Relation rel) {
 
         @Override
         public Void visitSyncFence(Relation syncFence) {
-            boolean idl = !context.useSATEncoding;
-            List<Event> allFenceSC = program.getThreadEvents(Event.class).stream()
-                    .filter(e -> e.hasTag(Tag.VISIBLE) && e.hasTag(Tag.FENCE) && e.hasTag(Tag.PTX.SC))
-                    .sorted(Comparator.comparingInt(Event::getGlobalId))
-                    .toList();
+            final boolean idl = !context.useSATEncoding;
+            final String relName = syncFence.getName().get(); // syncFence is base, it always has a name
+            List<Event> allFenceSC = program.getThreadEventsWithAllTags(VISIBLE, FENCE, PTX.SC);
+            allFenceSC.sort(Comparator.comparingInt(Event::getGlobalId));
```

Comment:
why do we need the sort?

Reply:
The original code used a sort, so I kept it. I think it is not necessary for correctness, but it makes sense to go over the events in that order.
That being said, the call to `program.getThreadEventsWithAllTags` should already return a sorted list, so the sort will do nothing (IIRC, java's sort is optimized to run in linear time on already sorted lists).
You can remove the call, if you want.

Reply:
Yes, the whole things boils down to call `e.getSuccessors()` so the ordering should be guaranteed (and even if not, it should not affect correctness). I will remove